### ISSUE TITLE
Migrate from Java EE to Jakarta EE 10

### ISSUE THREAD

**Issue by roman-mkh** [camunda-community-hub/camunda-platform-7-mail#151]
### Is your feature request related to a problem? Please describe.

[Wildfly 27](https://www.wildfly.org/news/2022/11/09/WildFly27-Final-Released/) adds support for [Jakarta EE 10](https://jakarta.ee/release/10/) and drops support for Jakarta EE 8.
According [Camunda Platform 7’s Road to Jakarta EE 10](https://camunda.com/blog/2023/06/camunda-platform-7-road-to-jakarta-ee-10/) with Camunda 7.19 running an WildFly 27 it is possible use only Jakarta EE 10 features.

### Describe the solution you'd like

Wee need replace Java EE class references and dependencies (javax.*) with Jakarta class references (jakarta.*)
May be it would be a good idea to introduce the 2-d artifact for it: 'camunda-bpm-mail-core-jakarta-x.x.x.jar'. (We keep 'camunda-bpm-mail-core-x.x.x.jar' for solution with Java EE).

### Describe alternatives you've considered

We can use [org.eclipse.transformer:transformer-maven-plugin](https://github.com/eclipse/transformer) to automate the migration. For easy transition we can even apply it direct to 'camunda-bpm-mail-core-x.x.x.jar' to produce 'camunda-bpm-mail-core-jakarta-x.x.x.jar' and provide it to maven repo under artifactId: camunda-bpm-mail-core-jakarta.

### Additional context

Reference: [ update from Camunda Platform 7.18.x to 7.19.0](https://docs.camunda.org/manual/latest/update/minor/718-to-719).

**Comment by roman-mkh** [comment]
@jonathanlukas unfortunately I can't manage it work with WildFly 27.0.1. I installed it according `shared-process-engine-wildfly.md` with the replacing curresponding dependency to jakarta.* . There is classloader issues. It works wenn the connector called synchronously from RESTfull service (over servlet), but has a fatal issue if it calls from Job (in Tx): `org.camunda.bpm.extension.mail.MailConnectorException: Failed to send mail: Provider for jakarta.activation.spi.MailcapRegistryProvider cannot be found`. Activation implementation module  `org.eclipse.angus.activation` and mail `org.eclipse.angus.mail` are under system modules of wildfly.
Any suggestions how to make it work?
_See full stacktrace in the attachment._
[MailConnectorException.txt](https://github.com/camunda-community-hub/camunda-platform-7-mail/files/12866173/MailConnectorException.txt)

**Comment by jonathanlukas** [comment]
@roman-mkh I looked into it and I was able to install the connector to `wildfly-27.0.1.Final`:

* Created file `server/wildfly-27.0.1.Final/modules/org/camunda/bpm/extension/camunda-bpm-mail-core/main/module.xml`:
```xml
<module xmlns="urn:jboss:module:1.0" name="org.camunda.bpm.extension.camunda-bpm-mail-core">
  <resources>
    <resource-root path="camunda-bpm-mail-core-jakarta-1.5.2-SNAPSHOT.jar" />
  </resources>

  <dependencies>
    <module name="jakarta.mail.api" />
    <module name="org.slf4j.slf4j-api" />
    <module name="jakarta.activation.api" />
    <module name="org.camunda.connect.camunda-connect-core" />
  </dependencies>
</module>
```
(please note I added jakarta-activation api as well, for some reason I needed to do this)
* Copied the `core-jakarta` file next to it.
* Made slf4j a module as described
* Added the mail connector module to the connector engine plugin module as described
* added the mail connector mdule to the process engine module as described
* exported the env variable `MAIL_CONFIG`

As base, I used the 7.19.0 Wildfly distro provided by Camunda.

Then, I deployed a project that calls the connector in a process. The connector is executed by the job executor. Beside from this log, everything works:
```
08:38:19,564 WARNING [jakarta.mail] (pool-8-thread-4) expected resource not found: /META-INF/javamail.default.address.map
```
Please double-check your setup

**Comment by roman-mkh** [comment]
@jonathanlukas thank You very much indeed that you took the problem seriously. 
I've downloaded camunda wildfly pre-packaged versions **'camunda-bpm-ee-wildfly-7.19.7-ee'** and **'camunda-engine-7.20.0-ee'** and installed them on Windows10 (java11). Additionaly I installed only **'camunda-bpm-mail-core-jakarta-1.5.2'** according your notes.
Unfortunately this does work only on synchronous calls. I prepared sample BPMN that reflect the problem. The first two activities "Send E-Mail 1" and "Send E-Mail 2" are working well, but "Send E-Mail 3" stuck until socket timeout (see stacktrace inthe attachment). (Database tx-timeout came before and created incident and process state cannot be stored any more. This cause that the process stucks forever).
Could you please run attached BPMN on your installation and check the behaviour? 
_you need to adjust 'to:' email-address (the variable "tx==true"can be used to skip synchronius sending emails)_
I observed the same behaviour on bothes 7.19 and 7.20.
Thank You for support!

![image](https://github.com/camunda-community-hub/camunda-platform-7-mail/assets/9946586/7958c16f-cb2c-4d2d-a128-23a83c0f724f)
[SendMail.zip](https://github.com/camunda-community-hub/camunda-platform-7-mail/files/12893046/SendMail.zip)
[socket-timeout.txt](https://github.com/camunda-community-hub/camunda-platform-7-mail/files/12893052/socket-timeout.txt)

**Comment by jonathanlukas** [comment]
Hi @roman-mkh , thank you for coming back and providing a concrete example.

I am able to reproduce the behaviour, but only if the process is deployed without a process application.

Given you package the process to a .war file and deploy it to the engine (as it is intended usually when using wildfly), there is no issue.

Still, I am not sure why the issue appears...

**Comment by roman-mkh** [comment]
Hi @jonathanlukas , thank you for rush answer. I will pack it to war and test and test again our real process with additional dependency to <module name="jakarta.activation.api" /> in the connector module.xml.
In mean while I modified (localy) camunda-bpm-mail-core to try use configured wildfy mail session (acoring manual [mail_subsystem](https://access.redhat.com/documentation/en-us/red_hat_jboss_enterprise_application_platform/7.0/html/configuration_guide/mail_subsystem#configuring_the_mail_subsystem). 
Unfortunately it gives me again 'java.lang.IllegalStateException: Not provider of jakarta.mail.util.StreamProvider was found'. I think this could be the root problem.

Anyway I found that such configuration could be useful to include in this connector. Here is my quick solution:
[MailServiceFactory.zip](https://github.com/camunda-community-hub/camunda-platform-7-mail/files/12894046/MailServiceFactory.zip) _(need new property 'mailsession.jndi.name' (example: mailsession.jndi.name=java:jboss/mail/MyMailSession) in mail-config.properties)_
